### PR TITLE
Feature/app 5493 create chart bar

### PR DESCRIPTION
**Pull request by acassiovilasboas**
## Description

- create bar chart
- fix translate tag status
- fix tag alpha color when status highlight
- add animate score chart
- add range color score chart

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## BarChart
Simulator Screenshot - iPhone 14 Pro - 2024-03-08 at 11 38 56

## Fix Tag
Simulator Screenshot - iPhone 14 Pro - 2024-03-08 at 11 39 05

## Update Score Chart
https://github.com/ocean-ds/ocean-ios/assets/47704536/d52c0003-03fa-44fb-adb1-5c921f586c82

## Checklist:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have made corresponding changes to the documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All new and existing tests passed

### CODE REVIEW

File: OceanComponents/Classes/Extensions UIKit/String+TagExtensions.swift [hunk 1]
```diff
@@ -14,10 +14,10 @@ public extension String {
         case "positive": return Ocean.Tag.Status.positive
         case "warning": return Ocean.Tag.Status.warning
         case "negative": return Ocean.Tag.Status.negative
-        case "complementary": return Ocean.Tag.Status.complementary
+        case "complementary", "highlight": return Ocean.Tag.Status.complementary
```
Comment (vromeiro) [inline OceanComponents/Classes/Extensions UIKit/String+TagExtensions.swift:17]: nao é no complementary, é. add um novo

case "highlight": return Ocean.Tag.Status.highlight

**Review by vromeiro — APPROVED** [review]

File: OceanComponents/Classes/Components SwiftUI/BarChart/OceanBarChartView.swift [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+import SwiftUI
+import OceanTokens
+import DGCharts
+
+struct OceanBarChartView: UIViewRepresentable {
+    public let entries: [BarChartModel]
+    public let color: UIColor
+    public let highlightColor: UIColor
+    public let isHighlightColor: Bool
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/BarChart/OceanBarChartView.swift:16]: Em inglês acabou ficando sem sentido, sugiro trocar para "shouldHightlightHighestValue"

File: OceanComponents/Classes/Components SwiftUI/BarChart/OceanBarChartView.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        for (index, model) in models.enumerated() {
+            let entry = BarChartDataEntry(x: Double(index), y: Double(model.value))
+            entries.append(entry)
+        }
+
+        return entries
+    }
+
+    private func determineBarColors() -> [UIColor] {
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/BarChart/OceanBarChartView.swift:58]: Como o método retorna, eu sugiro alterar para "getBarColors"

File: OceanComponents/Classes/Components SwiftUI/BarChart/OceanSwiftUI+BarChart.swift [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+extension OceanSwiftUI {
+
+    public class BarChartParameters: ObservableObject {
+        @Published public var title: String
+        @Published public var subtitle: String
+        @Published public var entries: [BarChartModel]
+        @Published public var color: UIColor
+        @Published public var highlightColor: UIColor
+        @Published public var isHighlight: Bool
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/BarChart/OceanSwiftUI+BarChart.swift:20]: Mesma coisa aqui, só pelo nome da variável fica difícil entender o que ela faz. Sugiro alterar para "shouldHighlightHighestValue"

File: OceanComponents/Classes/Components SwiftUI/BarChart/OceanSwiftUI+BarChart.swift [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+        public init(builder: Builder) {
+            self.init()
+            builder(self)
+        }
+
+        // MARK: View SwiftUI
+
+        public var body: some View {
+            headerView
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/BarChart/OceanSwiftUI+BarChart.swift:67]: Aqui é dúvida mesmo, funcionou colocando duas views direto sem colocar dentro de uma VStack?

**Review by renanmassaroto — APPROVED** [review]

File: OceanComponents/Classes/Components SwiftUI/BarChart/OceanSwiftUI+BarChart.swift (hunk 5 above)
Comment (acassiovilasboas) [reply OceanComponents/Classes/Components SwiftUI/BarChart/OceanSwiftUI+BarChart.swift:67]: Wow, vdd... funcionou sim.

**Review by renanmassaroto — APPROVED** [review]

**Review by vromeiro — APPROVED** [review]

**Pull request merged**